code/datums/dna.dm

```diff
@@ -272,6 +274,24 @@
 	return
 
 /////////////////////////// DNA MOB-PROCS //////////////////////
+/datum/dna/proc/update_body_size()
+	if(!holder)
+		return
+	var/desired_size
+	var/list/feat2num = list("Normal" = BODY_SIZE_NORMAL, "Tall" = BODY_SIZE_TALL, "Short" = BODY_SIZE_SHORT)
+	if(!species.default_features["body_size"])
+		desired_size = BODY_SIZE_NORMAL
+	else
+		desired_size ||= feat2num[features["body_size"]]
+
+	if(desired_size == current_body_size)
+		return
+
+	var/change_multiplier = desired_size / current_body_size
```

Comment:
What if target is big and then tries to become small?

Reply:
It works fine, proof provided in #development.

Reply:
It really seems like it doesn't work fine and just because the size is changing doesn't mean it's changing to the correct size.

Reply:
This is killing me so I went and checked and it does actually work right.